docs/0000-proposal-template.md

```diff
@@ -0,0 +1,43 @@
+# Feature name
```

Comment:
Could this be a github issue template instead? Once it "passes" by the clay team, it can be created as a markdown file.

Reply:
I think this fits into this conversation https://github.com/liferay/clay/pull/4126#discussion_r652737041